### PR TITLE
Apply database health to sap system

### PR DESCRIPTION
**Pull request by arbulu89**
# Description

Listen and apply database health changes in the sap system aggregate.

@abravosuse This implements a 1:1 feature parity with the current implementation. We will work in any other related feature in the future

## How was this tested?

Tested with UT and a new e2e

### CODE REVIEW

**Review by EMaksy — APPROVED** [review]
Hey man nice job, lgtm!

**Review by CDimonaco — APPROVED** [review]

File: test/trento/infrastructure/commanded/event_handlers/sap_system_database_health_event_handler_test.exs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  import Mox
+  import Trento.Factory
+
+  alias Trento.Databases.Events.DatabaseHealthChanged
+  alias Trento.Infrastructure.Commanded.EventHandlers.SapSystemDatabaseHealthEventHandler
+  alias Trento.SapSystems.Commands.UpdateDatabaseHealth
+
+  setup [:set_mox_from_context, :verify_on_exit!]
+
+  test "should dispatch DeregisterSapSystem commands when a database is deregistered" do
```
Comment (rtorrero) [inline test/trento/infrastructure/commanded/event_handlers/sap_system_database_health_event_handler_test.exs]: The title here seems wrong, missed copy paste replacement?

File: test/trento/sap_systems/sap_system_test.exs [hunk 2]
```diff
@@ -1060,6 +1075,55 @@ defmodule Trento.SapSystems.SapSystemTest do
...
+        build(:sap_system_registered_event,
+          sap_system_id: sap_system_id,
+          sid: sid
+        )
+      ]
+
+      assert_events_and_state(
+        initial_events,
+        [
+          UpdateDatabaseHealth.new!(%{
```
Comment (rtorrero) [inline test/trento/sap_systems/sap_system_test.exs:1103]: I'm probably missing something but why do we have here two identical UpdateDatabaseHealth to critical?

**Review by rtorrero — APPROVED** [review]
LGTM, only two minor comments

File: test/trento/infrastructure/commanded/event_handlers/sap_system_database_health_event_handler_test.exs (hunk 1 above)
Comment (arbulu89) [reply test/trento/infrastructure/commanded/event_handlers/sap_system_database_health_event_handler_test.exs]: yep, fixed thanks!

File: test/trento/sap_systems/sap_system_test.exs (hunk 2 above)
Comment (arbulu89) [reply test/trento/sap_systems/sap_system_test.exs:1103]: To see that the 2nd doesn't trigger a new event, as the health is the same

**Pull request merged**